pandas/conftest.py

```diff
@@ -1224,3 +1224,21 @@ def sort_by_key(request):
     Tests None (no key) and the identity key.
     """
     return request.param
+
+
+@pytest.fixture()
+def fsspectest():
+    pytest.importorskip("fsspec")
+    from fsspec.implementations.memory import MemoryFileSystem
+    from fsspec import register_implementation
+
+    class TestMemoryFS(MemoryFileSystem):
+        protocol = "testmem"
+        test = [None]
+
+        def __init__(self, **kwargs):
+            self.test[0] = kwargs.pop("test", None)
+            super().__init__(**kwargs)
+
+    register_implementation("testmem", TestMemoryFS, True)
```

Comment:
Keyword for `True`?

Reply:
Do you need to deregister the implementation as a teardown? Does any state leak between tests?

Reply:
Not really, but done now anyway. The tests were only seeing that the `.test[0]` changes, always with a new value.